go/logger/logger.go

```diff
@@ -12,14 +12,24 @@ type ExternalHandler interface {
 	Log(level keybase1.LogLevel, format string, args []interface{})
 }
 
-type Logger interface {
-	// Debug logs a message at debug level, with formatting args.
+type BaseLogger interface {
 	Debug(format string, args ...interface{})
+	Info(format string, args ...interface{})
+	Warning(format string, args ...interface{})
+	Error(format string, args ...interface{})
+}
+
+type CtxAndLogger interface {
```

Comment:
Since the logging objects we use on the client (logger.Logger) and the server (log.Logger) are different I thought using an interface with two different implementation would be a good way to go. I wrote the interface in a way that MetaContext already implements it on the client and we only need to write the server one. the stuff in context_with_logging.go is only used for testing.

Reply:
But are interfaces passed by value? For instance, `context.Context` isn't an interface for this reason, I think....

Reply:
But imagine this was the setup:

```go
type Context struct {
   logger logger.Logger
   ctx context.Context
}
```

then one liner:

```go
func (m MetaContext) ToLoggerContext() logger.Context {
  return logger.Context{logger : m.G().Log, ctx : m.ctx }
}```